compiler/code-gen/declarations.h

```diff
@@ -129,10 +130,14 @@ struct ClassDeclaration : CodeGenRootCmd {
   static void compile_deserialize(CodeGenerator &W, ClassPtr klass);
   static void compile_virtual_builtin_functions(CodeGenerator &W, ClassPtr klass);
 
+  static FunctionSignatureGenerator &&compile_class_method_signature(FunctionSignatureGenerator &&W, ClassPtr klass, std::string_view method_signature);
```

Comment:
I can't be 100% sure, but most of the signatures pass CodeGenerator by a simple reference, not a rvalue reference.
Is there a reason to pass it here by rvalue reference?

I'm also not sure there is a benefit in returning an object with rvalue reference (?). AFAIK, this place is not even that performance sensitive.